test/test_nn.py

```diff
@@ -12234,22 +12234,22 @@ def test_softshrink(self, device):
                            0.254, -0.24, -0.225, 0.104, 0.002, -0.001, 0.0574, 1.2344,
                            0.1748, -0.1797, -0.8125, 0.2051, -1.1328, 1.2344, -0.1562, 2.3554,
                            -0.1953, 0.0304, -0.3613, -1.3047, 1.0312, 0.1436, -0.6953, 0.5664,
-                           -0.5820, -0.3301, 0.8203, 0.6133, 0.5938],
+                           -0.5820, -0.3301, 0.8203, 0.6133, 0.5938, float('nan')],
```

Comment:
If we look at the top of this test case, it seems like it only tests cpu (see @onlyCPU), could we make sure the change is tested for other devices as well?


One possible way is to add a reference function via adding a `ref` argument to the `UnaryUfuncInfo` here

https://github.com/pytorch/pytorch/blob/e39955e82f494dcf6f22c0d81556e7cc26a281f7/torch/testing/_internal/common_methods_invocations.py#L16833

Reply:
I think above comment answers this one as well, not sure why this test is even here if we use functional way to test the op out in the firstplace(most of the testing is done in functional way)